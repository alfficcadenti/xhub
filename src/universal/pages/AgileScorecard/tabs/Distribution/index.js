import React, {useState, useEffect} from 'react';
import moment from 'moment';
import PieChart from '../../../../components/PieChart';
import LoadingContainer from '../../../../components/LoadingContainer';
import {formatPieData} from '../../utils';
import {checkResponse} from '../../../utils';

const Distribution = ({teams, from, to}) => {
    const [isLoading, setIsLoading] = useState(false);
    const [data, setData] = useState(false);
    const [error, setError] = useState();

    useEffect(() => {
        setIsLoading(true);
        const selectedTeam = teams.filter((x) => x?.checked && x?.name).map((x) => x?.name);
        const url = `/v1/score-card/distribution-work-data?from_date=${moment(from).format('YYYY-MM-DDTHH:mm:ss.sss[Z]')}&to_date=${moment(to).format('YYYY-MM-DDTHH:mm:ss.sss[Z]')}&team_names=${selectedTeam}`;
        fetch(url)
            .then(checkResponse)
            .then(setData)
            .catch(() => setError('Error loading the distribution of work. Try refreshing the page'))
            .finally(() => setIsLoading(false));
    }, [teams, from, to]);

    return (
        <LoadingContainer isLoading={isLoading} error={error}>
            <PieChart data={formatPieData(data)} title="" />
        </LoadingContainer>
    );
};

export default Distribution;