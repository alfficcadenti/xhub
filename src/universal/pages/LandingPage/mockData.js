// export const CSRData = [
//     {brandName: 'BEX', CSRTrend: [1, 2, 3, 6, 7, 3, 4, 10], CSRByDevice: {mobile: 90, tablet: 99, desktop: 60}},
//     {brandName: 'Hotels.com', CSRTrend: [1, 2, 3, 6, 5, 11, 10, 6], CSRByDevice: {mobile: 10, tablet: 99, desktop: 60}},
//     {brandName: 'Vrbo', CSRTrend: [10, 9, 3, 4, 5, 6, 4, 9], CSRByDevice: {mobile: 97, tablet: 50, desktop: 60}},
// ];

export const CSRData = [
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 0.99, Vrbo: 0.1, 'Hotels.com': 0.1,
    },
    {
        'time': '2020-05-21T11:30:00.000Z', BEX: 0.9, Vrbo: 0.1, 'Hotels.com': 0.2,
    },
    {
        'time': '2020-05-21T12:30:00.000Z', BEX: 0.9, Vrbo: 0.1, 'Hotels.com': 0.3,
    },
    {
        'time': '2020-05-21T13:30:00.000Z', BEX: 0.9, Vrbo: 1, 'Hotels.com': 0.7,
    },
    {
        'time': '2020-05-21T14:30:00.000Z', BEX: 0.9, Vrbo: 0.9, 'Hotels.com': 0.98,
    },
    {
        'time': '2020-05-21T15:30:00.000Z', BEX: 0.5, Vrbo: 0.7, 'Hotels.com': 0.8,
    },
    {
        'time': '2020-05-21T16:30:00.000Z', BEX: 0.99, Vrbo: 0.89, 'Hotels.com': 0.72,
    },
];

export const bookingsData = [
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 4000, Vrbo: 2400, 'Hotels.com': 2400,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 3000, Vrbo: 1398, 'Hotels.com': 2210,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 2000, Vrbo: 9800, 'Hotels.com': 2290,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 2780, Vrbo: 3908, 'Hotels.com': 2000,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 1890, Vrbo: 4800, 'Hotels.com': 2181,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 2390, Vrbo: 3800, 'Hotels.com': 2500,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 3490, Vrbo: 4300, 'Hotels.com': 2100,
    },
];