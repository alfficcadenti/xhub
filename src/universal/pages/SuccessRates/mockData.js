export const successRatesData = [
    [{'time': '2020-08-03T11:20:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': null}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': null}, {'brand': 'expedia', 'rate': null}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': null}, {'brand': 'wotif', 'rate': null}]}, {'time': '2020-08-03T11:25:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 86.70886075949366}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 97.33333333333334}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 73.13432835820896}, {'brand': 'wotif', 'rate': 75}]}, {'time': '2020-08-03T11:30:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 89.50617283950618}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.78048780487805}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': 66.66666666666666}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:35:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 84.57142857142857}, {'brand': 'cheaptickets', 'rate': 100}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.91304347826086}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 62.857142857142854}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:40:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 85.44303797468355}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.68085106382979}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': 68.42105263157895}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:45:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 88.30409356725146}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.25287356321839}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}],
    [{'time': '2020-08-03T11:20:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': null}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': null}, {'brand': 'expedia', 'rate': null}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': null}, {'brand': 'wotif', 'rate': null}]}, {'time': '2020-08-03T11:25:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 86.70886075949366}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 97.33333333333334}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 73.13432835820896}, {'brand': 'wotif', 'rate': 75}]}, {'time': '2020-08-03T11:30:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 89.50617283950618}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.78048780487805}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': 66.66666666666666}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:35:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 84.57142857142857}, {'brand': 'cheaptickets', 'rate': 100}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.91304347826086}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 62.857142857142854}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:40:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 85.44303797468355}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.68085106382979}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': 68.42105263157895}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:45:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 88.30409356725146}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.25287356321839}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}],
    [{'time': '2020-08-03T11:20:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': null}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': null}, {'brand': 'expedia', 'rate': null}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': null}, {'brand': 'wotif', 'rate': null}]}, {'time': '2020-08-03T11:25:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 86.70886075949366}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 97.33333333333334}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 73.13432835820896}, {'brand': 'wotif', 'rate': 75}]}, {'time': '2020-08-03T11:30:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 89.50617283950618}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.78048780487805}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': 66.66666666666666}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:35:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 84.57142857142857}, {'brand': 'cheaptickets', 'rate': 100}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.91304347826086}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 62.857142857142854}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:40:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 85.44303797468355}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.68085106382979}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': 68.42105263157895}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:45:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 88.30409356725146}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.25287356321839}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}],
    [{'time': '2020-08-03T11:20:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': null}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': null}, {'brand': 'expedia', 'rate': null}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': null}, {'brand': 'wotif', 'rate': null}]}, {'time': '2020-08-03T11:25:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 86.70886075949366}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 97.33333333333334}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 73.13432835820896}, {'brand': 'wotif', 'rate': 75}]}, {'time': '2020-08-03T11:30:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 89.50617283950618}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.78048780487805}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': 66.66666666666666}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:35:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 84.57142857142857}, {'brand': 'cheaptickets', 'rate': 100}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.91304347826086}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 62.857142857142854}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:40:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 85.44303797468355}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.68085106382979}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': 68.42105263157895}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:45:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 88.30409356725146}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.25287356321839}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}]
];
