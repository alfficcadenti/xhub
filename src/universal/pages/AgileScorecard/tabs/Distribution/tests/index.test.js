import React from 'react';
import {render, act} from '@testing-library/react';
import Distribution from '../index';
import {MOCK_TEAMS} from '../../../tests/mockData';
import '@testing-library/jest-dom';


describe('<Distribution />', () => {
    let wrapper = '';
    beforeEach(async () => {
        fetch.resetMocks();
        await act(async () => {
            wrapper = render(<Distribution teams={MOCK_TEAMS} from="2021-11-21" to="2021-11-22" />);
        });
    });

    it('renders Error message when api return error', async () => {
        fetch.mockImplementation(() => {
            return Promise.resolve({
                ok: false,
                json: () => Promise.resolve([])
            });
        });
        expect(wrapper.getByText(/Error loading the distribution of work/)).toBeInTheDocument();
        expect(wrapper.getByRole('alert')).toBeInTheDocument();
        expect(wrapper).toMatchSnapshot();
    });
});