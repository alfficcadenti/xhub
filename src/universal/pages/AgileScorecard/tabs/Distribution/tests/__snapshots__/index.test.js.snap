// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Distribution /> renders Error message when api return error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="loading-container "
      >
        <div
          aria-live="polite"
          class="Alert border--default in loading-alert"
          role="alert"
        >
          <div
            class="Alert__flair bg--color-bg-active"
          />
          <span
            class="SVGIcon SVGIcon--24px Alert__svg inline-svg text-muted"
          >
            <svg
              data-id="SVG_ALERT_WARNING__24"
              focusable="false"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
              >
                <path
                  d="M13 17.1a1 1 0 1 1-2 0 1 1 0 0 1 2 0"
                  fill="currentColor"
                  stroke="none"
                />
                <path
                  d="M10.6 4.5a1.6 1.6 0 0 1 2.8 0L21 17.7a1.6 1.6 0 0 1-1.4 2.5H4.4A1.6 1.6 0 0 1 3 17.7l7.6-13.2zM12 9V14"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </g>
            </svg>
          </span>
          <p>
            Error loading the distribution of work. Try refreshing the page
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="loading-container "
    >
      <div
        aria-live="polite"
        class="Alert border--default in loading-alert"
        role="alert"
      >
        <div
          class="Alert__flair bg--color-bg-active"
        />
        <span
          class="SVGIcon SVGIcon--24px Alert__svg inline-svg text-muted"
        >
          <svg
            data-id="SVG_ALERT_WARNING__24"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M13 17.1a1 1 0 1 1-2 0 1 1 0 0 1 2 0"
                fill="currentColor"
                stroke="none"
              />
              <path
                d="M10.6 4.5a1.6 1.6 0 0 1 2.8 0L21 17.7a1.6 1.6 0 0 1-1.4 2.5H4.4A1.6 1.6 0 0 1 3 17.7l7.6-13.2zM12 9V14"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </g>
          </svg>
        </span>
        <p>
          Error loading the distribution of work. Try refreshing the page
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
